rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/AbstractAppServicesClientImpl.java

```diff
@@ -0,0 +1,100 @@
+package com.redhat.service.bridge.rhoas;
+
+import java.util.function.Supplier;
+
+import com.openshift.cloud.api.kas.SecurityApi;
+import com.openshift.cloud.api.kas.auth.AclsApi;
+import com.openshift.cloud.api.kas.auth.TopicsApi;
+import com.redhat.service.bridge.rhoas.exceptions.AppServicesException;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.core.Vertx;
+
+abstract class AbstractAppServicesClientImpl {
```

Comment:
@kostola I personally find it quite hard to follow the intention of this class and would feel nervous about breaking it if I needed to refactor it. 

Some comments in the code about your general approach here would be useful IMHO.

Reply:
I recognize it's a bit complex. I commented the whole class now. :+1: